AsyncDisplayKit/ASMapNode.mm

```diff
@@ -206,15 +214,27 @@ - (void)takeSnapshot
                     UIGraphicsBeginImageContextWithOptions(image.size, YES, image.scale);
                     [image drawAtPoint:CGPointZero];
                     
-                    // Get a standard annotation view pin. Future implementations should use a custom annotation image property.
-                    MKAnnotationView *pin = [[MKPinAnnotationView alloc] initWithAnnotation:nil reuseIdentifier:@""];
-                    UIImage *pinImage = pin.image;
-                    CGSize pinSize = pin.bounds.size;
+                    UIImage *pinImage;
+                    CGPoint pinCenterOffset = CGPointZero;
+                    
+                    // Get a standard annotation view pin if there is no custom annotation block.
+                    if (!strongSelf.imageForStaticMapAnnotationBlock) {
+                      pinImage = [strongSelf.class defaultPinImageWithCenterOffset:&pinCenterOffset];
+                    }
                     
                     for (id<MKAnnotation> annotation in annotations) {
+                      if (strongSelf.imageForStaticMapAnnotationBlock) {
+                        // Get custom annotation view pin from custom annotation block.
+                        pinImage = strongSelf.imageForStaticMapAnnotationBlock(annotation, &pinCenterOffset);
+                        if (!pinImage) {
+                          // just for case block returned nil, which can happen
+                          pinImage = [strongSelf.class defaultPinImageWithCenterOffset:&pinCenterOffset];
+                        }
+                      }
+                      
```

Comment:
Maybe this can be changed to:

Always get the default pin, just in case it might be used (better here than in a for loop)

```
CGPoint defaultPinCenterOffset = CGPointZero;
UIImage *defaultImage = [strongSelf.class defaultPinImageWithCenterOffset:&defaultPinCenterOffset];
```

Now that we have the default image, we can loop through the annotations, if the property is not set or if it returns nil, we will just use the default pin. 
```
for (id<MKAnnotation> annotation in annotations) {
  UIImage *pinImage = nil;
  CGPoint pinCenterOffset = CGPointZero;

  if (strongSelf.imageForStaticMapAnnotationBlock) {
    pinImage = strongSelf.imageForStaticMapAnnotationBlock(annotation, &pinCenterOffset);
  }

  if (!pinImage) {
    pinImage = defaultImage;
    pinCenterOffset = defaultPinCenterOffset;
  }

[...]
```

The idea is that most likely if we are to use the defaultImage it is most likely going to be for more than one annotation, and this way we don't have to call the defaultPin several times since it will always return the same value anyway.

What do you think?

PS: If we won't go this way, either remove the comments that you have, or change the references from 'annotation view' to 'image for pin' or something to be consistent with the new code.